nbclient/util.py

```diff
@@ -4,9 +4,52 @@
 # Distributed under the terms of the Modified BSD License.
 
 import asyncio
+import sys
 import inspect
 
 
+def check_ipython():
+    # original from vaex/asyncio.py
+    IPython = sys.modules.get('IPython')
+    if IPython:
+        IPython_version = tuple(map(int, IPython.__version__.split('.')))
+        if IPython_version < (7, 0, 0):
+            raise RuntimeError(f'You are using IPython {IPython.__version__} while we require'
+                               '7.0.0+, please update IPython')
+
+
+def check_patch_tornado():
+    """If tornado is imported, add the patched asyncio.Future to its tuple of acceptable Futures"""
+    # original from vaex/asyncio.py
+    if 'tornado' in sys.modules:
+        import tornado.concurrent
+        if asyncio.Future not in tornado.concurrent.FUTURES:
+            tornado.concurrent.FUTURES = tornado.concurrent.FUTURES + (asyncio.Future, )
+
+
+def just_run(coro):
+    """Make the coroutine run, even if there is an event loop running (using nest_asyncio)"""
+    # original from vaex/asyncio.py
+    try:
+        loop = asyncio.get_event_loop()
+        had_loop = True
+    except RuntimeError:
+        loop = None
+        had_loop = False
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+    try:
+        if had_loop:
+            check_ipython()
+            import nest_asyncio
+            nest_asyncio.apply()
+            check_patch_tornado()
+        return loop.run_until_complete(coro)
+    finally:
+        if not had_loop:  # remove loop if we did not have one
+            asyncio.set_event_loop(None)
```

Comment:
Why not keep the event loop we just created?

Reply:
To keep things as they were, otherwise you leave a non-running loop.

Reply:
On the other hand we are probably going to recreate an event loop in a subsequent call. I don't think having an idle event loop is a problem?

Reply:
I *think* we need to check if the existing loop is then non-running, and make it run again? I thought this would be safer, a clean start each time, e.g. no old stuff leftover because of obscure bugs.

Also, this solution has survived quite a while in vaex' CI, so I would be reluctant to change this unless we have good test coverage, which would make me really happy :)

So not against it, just being conservative here :)

Reply:
:+1: